src/commands/dev/dev.mjs

```diff
@@ -55,16 +54,26 @@ import { createDevExecCommand } from './dev-exec.mjs'
 const netlifyBuildPromise = import('@netlify/build')
 
 const startStaticServer = async ({ settings }) => {
-  const server = new StaticServer({
-    rootPath: settings.dist,
-    name: 'netlify-dev',
-    port: settings.frameworkPort,
-    templates: {
-      notFound: path.join(settings.dist, '404.html'),
+  const server = Fastify()
+  server.register(fastifyStatic, {
+    root: settings.dist,
+    // eslint-disable-next-line no-unused-vars
+    setHeaders: (res, _path, _stat) => {
+      res.setHeader('X-Powered-by', 'netlify-dev')
     },
   })
-
-  await promisify(server.start.bind(server))()
+   
+  server.setNotFoundHandler((_req, res) => {
+    res.sendFile('404.html', settings.dist)
+  })
+  server.addHook('onRequest', (req, reply) => {
+    const validMethods = ['GET', 'HEAD']
+    if (!validMethods.includes(req.method)) {
+      reply.code(405).send()
+      return reply
+    }
```

Comment:
Adapted from https://github.com/nbluis/static-server/blob/84999917193f664c205efd587934bc6ea361fa78/server.js#L163-L164 since the `should rewrite POST request if content-type is missing and not crash dev server` test ([source](https://github.com/netlify/cli/blob/e3db3144d77f7a3c7abc9b2fe803d701242492f4/tests/integration/0.command.dev.test.cjs#L240-L241)) depends on the correct status code.